src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```
@@ -34,14 +35,16 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @PutMapping("/movie-sessions")
-    public void addToCart(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void addToCart(Authentication authentication, @RequestParam Long movieSessionId) {
+        String email = authentication.getPrincipal().toString();
         shoppingCartService.addSession(
-                movieSessionService.get(movieSessionId), userService.get(userId));
+                movieSessionService.get(movieSessionId), userService.findByEmail(email).get());
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
-        User user = userService.get(userId);
+    public ShoppingCartResponseDto getByUser(Authentication authentication) {
+        String email = authentication.getPrincipal().toString();
+        User user = userService.findByEmail(email).get();
```

let's throw an exception here in case optional is null